src/Ocean/Domains/rectangular_domain.jl

```diff
@@ -0,0 +1,194 @@
+using MPI
+
+using ClimateMachine.Mesh.Grids:
+    DiscontinuousSpectralElementGrid, polynomialorder
+
+using ClimateMachine.Mesh.Topologies: StackedBrickTopology
+
+#####
+##### RectangularDomain
+#####
+
+struct RectangularDomain{FT, G} <: AbstractDomain
```

Comment:
Docs?

Reply:
Do we put docs on types and constructors or just constructors? Over at Oceananigans we put docs on everything (types and constructors) but my thought these days is that docs on constructors really pollutes the help messages and makes it harder to use stuff. Happy to doc this all up though.

Reply:
My immediate reaction is that you want to document the constructors, not necessarily the type. I'm not entirely sure though, because Greg's point makes sense. So, when in doubt, I look at what the Julia docs do. [Here](https://github.com/JuliaLang/julia/blob/788b2c77c10c2160f4794a4d4b6b81a95a90940c/stdlib/Sockets/src/IPAddr.jl#L31-L41) is one example where they have a docstring on one constructor only. But [here](https://github.com/JuliaLang/julia/blob/788b2c77c10c2160f4794a4d4b6b81a95a90940c/stdlib/SparseArrays/src/sparsevector.jl#L12-L16), they document the type.

So I guess the answer is: use your best judgement? @simonbyrne: thoughts?

Reply:
That's fine with me. I think we've been using mixed styles, but perhaps it's context dependent.